### PR TITLE
docs: add provider to the documentation

### PR DESCRIPTION
**Pull request by beeme1mr**
## This PR

- adds a section on the install page that lists the available providers
- includes the readmes of the available providers

### How to test

- Install: https://deploy-preview-1027--polite-licorice-3db33c.netlify.app/installation/
- Overview Page: https://deploy-preview-1027--polite-licorice-3db33c.netlify.app/providers/
- Go: https://deploy-preview-1027--polite-licorice-3db33c.netlify.app/providers/go/
- Java: https://deploy-preview-1027--polite-licorice-3db33c.netlify.app/providers/java/
- Node.JS: https://deploy-preview-1027--polite-licorice-3db33c.netlify.app/providers/nodejs/
- PHP: https://deploy-preview-1027--polite-licorice-3db33c.netlify.app/providers/php/
- .NET: https://deploy-preview-1027--polite-licorice-3db33c.netlify.app/providers/dotnet/
- Web: https://deploy-preview-1027--polite-licorice-3db33c.netlify.app/providers/web/

### CODE REVIEW

**Comment by beeme1mr** [comment]
Web isn't working correctly. Looking into it.

File: docs/installation.md [hunk 1]
```diff
@@ -49,32 +49,3 @@ brew install flagd
 ```shell
 go install github.com/open-feature/flagd/flagd@latest
 ```
-
```
Comment (Kavindu-Dodan) [inline docs/installation.md:52]: +1 for removing this section

Comment (Kavindu-Dodan) [reply docs/installation.md:52]: I think we should include a cross-link in this installation page for providers

ex:- `Once flagd is installed, you can evaluate feature flags using your favorite language using flagd provider[Link to provider page] `

**Review by Kavindu-Dodan — APPROVED** [review]
Changeset looks good. I left a [comment](https://github.com/open-feature/flagd/pull/1027#discussion_r1449520172) on potential improvement

File: docs/installation.md (hunk 1 above)
Comment (beeme1mr) [reply docs/installation.md:52]: Done.

**Review by Kavindu-Dodan — APPROVED** [review]

**Pull request merged**